### DISCUSSION TITLE
Customizing model

### DISCUSSION THREAD

**Discussion by FerseinChris** [Project-MONAI/MONAI#7252] [Q&A]
While I respect MONAI as a platform, I find it quite difficult to try to improve the model. Is there a method for really understanding the code written by MONAI developers?  Let's say I want to separate every part of UNet, to have the code only for encoder, for decoder and skip connexions.

**Comment by KumoLiu** [comment]
Hi @FerseinChris, thanks for your interest here.
You can get the middle part like this:
```
from monai.networks.nets import UNet

net = UNet(spatial_dims=3, in_channels=1, out_channels=2, channels=(32,64,128,256), strides=(2,2,2))
print(net.model[1])
```

You can also refer to this [discussion](https://github.com/Project-MONAI/MONAI/discussions/5796) about how to get the intermediate feature.
Also FYI https://github.com/Project-MONAI/tutorials/blob/main/modules/UNet_input_size_constraints.ipynb

**Reply by FerseinChris** [reply]
Hi! I have a small question: calling net.model[1], does net.model[1] means the bottleneck of unet model? And if I want to call the encoder part i just have to call net.model[0]. Is it ok?

**Reply by FerseinChris** [reply]
Also How can I get output features of layers?
